src/PluginsHelper.php

```diff
@@ -131,4 +139,281 @@ public static function get_plugin_data( $plugin ) {
+				self::set_activation_status( $job_id, 'failed' );
+			}
+			return new \WP_Error( 'woocommerce_plugins_invalid_plugins', __( 'Plugins must be a non-empty array.', 'woocommerce-admin' ), 404 );
+		}
+
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		// the mollie-payments-for-woocommerce plugin calls `WP_Filesystem()` during it's activation hook, which crashes without this include.
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+
+		/**
+		 * Filter the list of plugins to activate.
+		 *
+		 * @param array $plugins A list of the plugins to activate.
+		 */
+		$plugins = apply_filters( 'woocommerce_admin_plugins_pre_activate', $plugins );
+
+		$plugin_paths      = self::get_installed_plugins_paths();
+		$errors            = new \WP_Error();
+		$activated_plugins = array();
+
+		foreach ( $plugins as $plugin ) {
+			$slug = $plugin;
+			$path = isset( $plugin_paths[ $slug ] ) ? $plugin_paths[ $slug ] : false;
+
+			if ( ! $path ) {
+				$errors->add(
+					$plugin,
+					/* translators: %s: plugin slug (example: woocommerce-services) */
+					sprintf( __( 'The requested plugin `%s`. is not yet installed.', 'woocommerce-admin' ), $slug )
+				);
+				continue;
+			}
+
+			$result = activate_plugin( $path );
+			if ( ! is_null( $result ) ) {
+				do_action( 'woocommerce_plugins_activate_error', $slug, $result );
+
+				$errors->add(
+					$plugin,
+					/* translators: %s: plugin slug (example: woocommerce-services) */
+					sprintf( __( 'The requested plugin `%s` could not be activated.', 'woocommerce-admin' ), $slug )
+				);
+				continue;
+			}
+
+			$activated_plugins[] = $plugin;
+		}
+
+		$data = array(
+			'activated' => $activated_plugins,
+			'active'    => self::get_active_plugin_slugs(),
+			'errors'    => $errors,
+		);
+
+		if ( $job_id ) {
+			self::set_activation_status( $job_id, 'complete', $data );
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Schedule plugin activation.
+	 *
+	 * @param array $plugins Plugins to activate.
+	 * @return string Job ID.
+	 */
+	public static function schedule_activate_plugins( $plugins ) {
+		if ( empty( $plugins ) || ! is_array( $plugins ) ) {
+			return new \WP_Error( 'woocommerce_plugins_invalid_plugins', __( 'Plugins must be a non-empty array.', 'woocommerce-admin' ), 404 );
+		}
+
+		$job_id = uniqid();
+		self::set_activation_status( $job_id, 'pending' );
+		WC()->queue()->schedule_single( time() + 5, 'woocommerce_plugins_activate_callback', array( $plugins, $job_id ) );
+
+		return $job_id;
+	}
+
+	/**
+	 * Set the status of an installation job.
+	 *
+	 * @param string $id Job ID.
+	 * @param string $status Status.
+	 * @param array  $data Job data.
+	 */
+	public static function set_installation_status( $id, $status, $data = array() ) {
+		$jobs        = get_transient( 'woocommerce_plugins_installation_status', array() );
+		$jobs[ $id ] = array(
+			'status' => $status,
+			'data'   => $data,
+		);
+		set_transient( 'woocommerce_plugins_installation_status', $jobs );
```

Comment:
Will saving this data in transients be necessary? Could you also make use of the scheduled actions instead, by using this [search functionality](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/queue/class-wc-action-queue.php#L136-L159):
`WC()->queue()->search( array( 'hook' => 'woocommerce_plugins_install_callback' ) )`?

Reply:
I would have also like to use AS directly to handle job status, however, I don't think it's well supported or encouraged.

I pushed what this would look like in https://github.com/woocommerce/woocommerce-admin/pull/8079/commits/d2c832cd3e29be52903300ca2d5bd56a9a2f63b2.  As far as I'm aware, there's no supported API for retrieving details of an action, though it is possible to make multiple searches by hook/IDs and guess the status of an action.

I believe the recommendation is also not to log any information to AS that may be needed later, so transients allow a bit more flexibility on this front.

But let me know if you know a way around this or had something else in mind and I can adjust.

Reply:
> As far as I'm aware, there's no supported API for retrieving details of an action, though it is possible to make multiple searches by hook/IDs and guess the status of an action.
> I believe the recommendation is also not to log any information to AS that may be needed later, so transients allow a bit more flexibility on this front.

Gotcha, I trust your word for it, as I haven't played around with it myself. In that case I think transients will do the trick.

Reply:
>I was thinking instead of searching for a specific action to do the search by the hook, something like this:

I'm a bit against just searching by hook since over the lifetime of the site this could result in a large result set.  We can however use both hooks and the unique ID to keep the filtering at the DB level.

> it appears we would have to check for the Class type, given a finished action has type ActionScheduler_FinishedAction, although it does appear to provide a is_finished function on the object:

Nice find on `is_finished()`!  Unfortunately, this does not provide us with enough for some status types (for example, `failed`).  I also have (minor) concerns that checking against so many class types could be brittle with logic changes in AS.

I did manage to find a way to retrieve the status directly from the store using `ActionScheduler_DBStore`.  This does add a single extra query (assuming it has not already been hit), but I prefer using the AS status as the source of truth.

Let me know if this looks alright to you.

>We could try to use this, and if we think we need more information that is not part of the Action args, we make use of transients on top of this

I'm not opposed to trying this out first.  I would argue for use of the action data or transients and not a combination, but given this data is managed by the PluginHelper, we can easily swap this later if needed.